pkg/client/request_test.go

```diff
@@ -834,7 +834,9 @@ func TestDoRequestNewWayReader(t *testing.T) {
 		t.Errorf("Expected: %#v, got %#v", expectedObj, obj)
 	}
 	tmpStr := string(reqBodyExpected)
-	fakeHandler.ValidateRequest(t, "/api/v1beta1/foo/bar/baz?labels=name%3Dfoo&timeout=1s", "POST", &tmpStr)
+	requestURL := testapi.ResourcePathWithPrefix("foo", "bar", "", "baz")
```

Comment:
Renamed the function to ResourcePathWithPrefixAndNamespaceQuery to remove confusion.